### ISSUE TITLE
Updating Tesseract?

### ISSUE THREAD

**Issue by timofonic** [koreader/koreader-base#361]
Hello.

What about updating Tesseract? I did read they improved accuracy somehow.

Kind regards.

**Comment by houqp** [comment]
Good catch, we have been lagging behind for quiet a bit now. Bumped in https://github.com/koreader/koreader-base/pull/360

**Comment by houqp** [comment]
nvm, i reverted the version bump. some of our own patch is causing conflicts. @chrox can you take a look? I forgot why we added tesseract_mod for libk2pdfopt.

**Comment by houqp** [comment]
Ha, i found the diff:

```diff
+#include "config_auto.h"
 ///////////////////////////////////////////////////////////////////////
 // File:        tessdatamanager.cpp
 // Description: Functions to handle loading/combining tesseract data files.
@@ -37,9 +38,12 @@ bool TessdataManager::Init(const char *data_file_name, int debug_level) {
   debug_level_ = debug_level;
   data_file_ = fopen(data_file_name, "rb");
   if (data_file_ == NULL) {
-    tprintf("Error opening data file %s\n", data_file_name);
-    tprintf("Please make sure the TESSDATA_PREFIX environment variable is set "
-            "to the parent directory of your \"tessdata\" directory.\n");
+    if (debug_level_)
+        {
+        tprintf("Error opening data file %s\n", data_file_name);
+        tprintf("Please make sure the TESSDATA_PREFIX environment variable is set "
+                "to the parent directory of your \"tessdata\" directory.\n");
+        }
     return false;
   }
   fread(&actual_tessdata_num_entries_, sizeof(inT32), 1, data_file_);
```

Looks like only the `include "config_auto.h"` is necessary?

**Comment by houqp** [comment]
latest tesseract upstream already has the basapi.cpp and configure.ac patch

**Comment by chrox** [comment]
The `tesseract_mod` patches are originally in Willus's k2pdfopt. They defined several user-friendly C APIs (wrapping up Tesseract's C API) for doing OCR in k2pdfopt. Since the dependencies of Tesseract and Leptonica are in upstream K2pdfopt, we can only update Tesseract and Leptonica after K2pdfopt has updated them.

**Comment by timofonic** [comment]
Anyway, I sent a forum message.

Does the developer uses Git? Maybe if he uses it, k2pdfopt deployment could be easier :)

http://www.mobileread.com/forums/showthread.php?p=3235510#post3235510

I opened this too. I'm confused about where to "complain".
https://github.com/koreader/libk2pdfopt/issues/26

**Comment by Frenzie** [comment]
I believe this should be fixed in #420, so closing now.
